ui/dashboards/src/context/DashboardProvider/panel-editor-slice.ts

```diff
@@ -223,8 +227,13 @@ export function createPanelEditorSlice(): StateCreator<
         },
       };
 
-      // Open the editor with the new state
       set((state) => {
+        // Add the new panel group if one was created for the panel
+        if (newGroup !== undefined) {
+          addPanelGroup(state, newGroup);
+        }
+
+        // Open the editor with the new state
         state.panelEditor = editorState;
       });
```

Comment:
Could be cool to always say either `set((state) => ...)` or `set((draft) => ...)` (I noticed we have these different argument names when calling `set` in in this file and panel-group-editor-slice)

I lean toward `state` but also it's not a big deal